package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +554,79 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
-        if not iterable(self.weights) and self.weights == "mass":
-            pass
-        else:
-            self.weights = get_weights(self.mobile_atoms, self.weights)
 
-        # cannot use arbitrary weight array (for superposition) with
-        # groupselections because arrays will not match
-        if (len(self.groupselections) > 0 and (
-                iterable(self.weights) or self.weights not in ("mass", None))):
-            raise ValueError("groupselections can only be combined with "
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
+        # check weights type
+
+        if self.weights is None:
+            pass
+        elif (type(self.weights) == str) or \
+            (np.array(self.weights).ndim == 1) and (np.array(self.weights).dtype 
+                                                in (np.dtype('float64'),np.dtype('int64'))):
```

Comment:
You could combine these using `if not iterable(self.weights) or np.array( ...` to avoid needing the separate pass statement.